Harden-Windows-Security Module/Main files/Core/Protect-WindowsSecurity.psm1

```diff
@@ -31,36 +31,109 @@ Function Protect-WindowsSecurity {
                 $true
             })]
         [System.String[]]$Categories
-    )
-
-    # This class provides a list of valid values for the Categories parameter of the Protect-WindowsSecurity function
-    Class Categoriex : System.Management.Automation.IValidateSetValuesGenerator {
-        [System.String[]] GetValidValues() {
-            $Categoriex = @(
-                'WindowsBootManagerRevocations',
-                'MicrosoftSecurityBaselines',
-                'Microsoft365AppsSecurityBaselines',
-                'MicrosoftDefender',
-                'AttackSurfaceReductionRules',
-                'BitLockerSettings',
-                'TLSSecurity',
-                'LockScreen',
-                'UserAccountControl',
-                'WindowsFirewall',
-                'OptionalWindowsFeatures',
-                'WindowsNetworking',
-                'MiscellaneousConfigurations',
-                'WindowsUpdateConfigurations',
-                'EdgeBrowserConfigurations',
-                'CertificateCheckingCommands',
-                'CountryIPBlocking',
-                'DownloadsDefenseMeasures',
-                'NonAdminCommands'
-            )
-            return [System.String[]]$Categoriex
+)
+
+    DynamicParam {
+
+        # Add the dynamic parameters to the param dictionary
+        $ParamDictionary = [System.Management.Automation.RuntimeDefinedParameterDictionary]::new()
+
+        if ('MicrosoftSecurityBaselines' -in $PSBoundParameters['Categories']) {
+
+            # Create a dynamic parameter for -MicrosoftSecurityBaselines_NoOverrides
+            $MicrosoftSecurityBaselines_NoOverrides = [System.Management.Automation.ParameterAttribute]@{
+                Mandatory        = $false
+                ParameterSetName = 'Categories'
+                HelpMessage      = 'Apples the Microsoft Security Baselines without the optional overrides'
```

Comment:
I believe this should say `Applies the...` instead of `Apples the...`?

Reply:
You're right, will fix it, thanks :)